geniza/corpus/tests/test_corpus_views.py

```diff
@@ -202,6 +204,29 @@ def test_get_context_data(self, rf):
         context_data = docsearch_view.get_context_data()
         assert context_data["total"] == 22
 
+    def test_shelfmark_boost(self, document, multifragment):
+        # integration test for shelfmark field boosting
+        # in solr configuration
+
+        # create a second document with a different shelfmark
+        # that references the shelfmark of the first in the description
+        related_doc = Document.objects.create(
+            description="See also %s" % document.shelfmark
+        )
+        TextBlock.objects.create(document=related_doc, fragment=multifragment)
+        # wait to make sure solr index is updated with document fixtures
+        sleep(1)
```

Comment:
Will this sleep command guarantee the necessary amount of wait time in all environments? I wonder if there could be a way to run a loop checking for the fixture's presence in the index instead of sleeping—though I'm not familiar enough with solr to know how doable that is.

Reply:
Good question — we have the test solr configured with a pretty aggressive commit time, you can see it here https://github.com/Princeton-CDH/geniza/blob/main/geniza/settings/components/base.py#L107-L110 

So I think it _should_ be enough, but there have been cases in the past where things fail — e.g., a test that works locally but fails in CI. This is the approach we've been using in other projects, but even having the sleep isn't ideal, it would probably be better to use a solr commit that would ensure the index is current.

I'll see if I can add a solr commit here instead.